### PR TITLE
Add initial documentation and instances

### PR DESCRIPTION
**Pull request by thomashoneyman**
**Description of the change**
This update adds module documentation, a README example, expands on the instances available in the library, and adds a couple of essential helper functions. I've tried to keep things as minimal as possible for the time being; we can always expand functionality over time.

### CODE REVIEW

**Comment by thomashoneyman** [comment]
@garyb Once this looks good I can use it to release v1.0.0 of the library (or v0.1.0 if you'd prefer).

**Review by garyb — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 [![Maintainer: garyb](https://img.shields.io/badge/maintainer-garyb-teal.svg)](https://github.com/garyb)
 [![Maintainer: thomashoneyman](https://img.shields.io/badge/maintainer-thomashoneyman-teal.svg)](https://github.com/thomashoneyman)
 
-The library summary hasn't been written yet (contributions are welcome!). The library summary describes the library's purpose in one to three sentences.
+Utilities for creating and managing push/pull subscriptions, inspired by the [event](https://github.com/paf31/purescript-event) library. This library is used to implement subscriptions in [Halogen](https://github.com/purescript-halogen/purescript-halogen), but it can be used independently of Halogen.
```
Comment (garyb) [inline README.md]: Oh wait, actually, this library just provides the push part :) behaviours are the pull part of push-pull FRP, AFAIK.

**Review by garyb — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (thomashoneyman) [reply README.md]: Updated!

**Review by garyb — APPROVED** [review]
:+1:

**Pull request merged**